internal/services/openai.go

```diff
@@ -69,12 +73,25 @@ func (o openAI) askChatGPT(body io.Reader) (*ChatGPTResponse, error) {
 	req.Header.Set("Content-Type", "application/json")
 	req.Header.Set("Authorization", "Bearer "+o.token)
 
+	var currentReqResponseTime time.Duration
+	trace := &httptrace.ClientTrace{
+		GotFirstResponseByte: func() {
+			currentReqResponseTime = time.Since(time.Now())
+		},
+	}
+	req = req.WithContext(httptrace.WithClientTrace(req.Context(), trace))
+
 	resp, err := http.DefaultClient.Do(req)
 	if err != nil {
 		return nil, err
 	}
 
 	defer resp.Body.Close()
+
+	appmetrics.OpenAIResponseTimeOps.With(prometheus.Labels{
+		"api_url": o.chatGptURL,
```

Comment:
Status seems okay, but I would just drop this `api_url` label. It never changes—if it ever does we can add some stuff later.